controllers/propagator/policy_controller.go

```diff
@@ -122,20 +123,63 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 				return reconcile.Result{}, err
 			}
 
+			if len(replicatedPlcList.Items) == 0 {
+				log.V(2).Info("No replicated policies to delete.")
+
+				return reconcile.Result{}, nil
+			}
+
+			policiesChan := make(chan policiesv1.Policy, len(replicatedPlcList.Items))
+			deletionResultsChan := make(chan deletionResult, len(replicatedPlcList.Items))
+
+			var numWorkers int
+			if len(replicatedPlcList.Items) > concurrencyPerPolicy {
+				numWorkers = concurrencyPerPolicy
+			} else {
+				numWorkers = len(replicatedPlcList.Items)
+			}
+
+			for i := 0; i < numWorkers; i++ {
+				go plcDeletionWrapper(r, policiesChan, deletionResultsChan)
+			}
+
+			log.V(2).Info("Scheduling work to handle deleting replicated policies for the policy", "policy",
+				instance.GetName())
+
 			for _, plc := range replicatedPlcList.Items {
-				log := log.WithValues("name", plc.GetName(), "namespace", plc.GetNamespace())
+				policiesChan <- plc
+			}
+
+			// Wait for all the deletions to be processed.
+			log.V(2).Info("Waiting for the result of deleting the replicated policies", "count", len(policiesChan))
```

Comment:
Optional: I'd like this as a log level of 1 since waiting can take a while for many replicated policies.